error/s2n_errno.c

```diff
@@ -270,7 +274,8 @@ static const char *no_such_error = "Internal s2n error";
     ERR_ENTRY(S2N_ERR_SECRET_SCHEDULE_STATE, "Correct inputs to secret calculation not available") \
     ERR_ENTRY(S2N_ERR_LIBCRYPTO_VERSION_NUMBER_MISMATCH, "The libcrypto major version number seen at compile-time is different from the major version number seen at run-time") \
     ERR_ENTRY(S2N_ERR_LIBCRYPTO_VERSION_NAME_MISMATCH, "The libcrypto major version name seen at compile-time is different from the major version name seen at run-time") \
-    ERR_ENTRY(S2N_ERR_CERT_OWNERSHIP, "The ownership of the certificate chain is incompatible with the operation")
+    ERR_ENTRY(S2N_ERR_CERT_OWNERSHIP, "The ownership of the certificate chain is incompatible with the operation") \
+    ERR_ENTRY(S2N_ERR_LIBCRYPTO_ERROR, "An internal error has occurred in the libcrypto API")
 /* clang-format on */
```

Comment:
I don't love how generic this error is (I worry people will use it when they should use a more specific one) but I'm also not sure what else you could use. You only use it places we really don't expect libcrypto calls to fail. Maybe make that more clear? S2N_ERR_INTERNAL_LIBCRYPTO_ERROR?

But feel free to ignore this comment. What you're doing is fine.